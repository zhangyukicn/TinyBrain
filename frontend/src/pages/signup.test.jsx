import React from 'react';
import { getDefaultNormalizer, render, screen } from '@testing-library/react';
import { configure, EnzymeAdapter, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Lobby from './lobby';
import Signup from './signup';
configure({ adapter: new Adapter() })

describe('SignUp', () => {
    const noop = () => { };

    const modalStarta = shallow(<Signup />);

    const props = {
      username: 'Yuki',
      passwords: 'yyyyyyyyyy',
      mail: 'zzdfhjkdjfh@gmail.com'
    }

    it('is matched with snapshot', () => {
      expect(modalStarta).toMatchSnapshot();
    });

    it('triggers submitSignUp onClike event handler when clicked', () => {
      const onClick = jest.fn();
      shallow(<submitSignUp onClick={onClick} />).simulate('click');
      expect(onClick).toHaveBeenCalledTimes(1);
    })

    it('triggers toLogIn onClike event handler when clicked', () => {
      const onClick = jest.fn();
      shallow(<toLogIn onClick={onClick} />).simulate('click');
      expect(onClick).toHaveBeenCalledTimes(1);
    })

    it('should call checkPassword prop with input value', () => {
      const onChange = jest.fn();
      const component = shallow(<checkPassword onChange={onChange} />);
      component.simulate('change', { target: { value: 'zyruieriewqr' } });
      expect(component).toBeDefined();
    })
  })
