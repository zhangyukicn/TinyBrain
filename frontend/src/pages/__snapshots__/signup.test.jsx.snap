// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp is matched with snapshot 1`] = `ShallowWrapper {}`;
